src/mp_integer.hpp

```diff
@@ -2294,6 +2294,21 @@ class mp_integer
 		{
 			return mpz_view(*this);
 		}
+		/// Get an \p non-const mpz_ptr to dynamic storage of \p this.
+		/**
+		 * This method will return a pointer to the dynamic storage, mpz_struct_t. If \p this is currently stored in static
+		 * storage, \p this will be promoted to dynamic storage first
+		 *
+		 * Note that \p this will remain stored in dynamic storage even if after operations on \p this, value of \p this can
+		 * be stored in static storage.
+		 *
+		 * @return an \p non-const mpz_ptr to dynamic storage of \p this.
+		 */
+		mpz_ptr get_mpz_ptr()
```

Comment:
Would it be possible to rename the method to ``_get_mpz_ptr()`` (leading underscore)? There's a convention in Piranha that methods starting with ``_`` are considered potentially dangerous. E.g., here a user could call ``mpz_clear()`` on the output pointer, and then the destructor of ``this`` would also call ``mpz_clear()``, leading to undefined behaviour.